src/databricks/labs/ucx/aws/access.py

```diff
@@ -52,30 +55,41 @@ def list_uc_roles(self, *, single_role=True, role_name="UC_ROLE", policy_name="U
         """
         roles: list[AWSUCRoleCandidate] = []
         missing_paths = self._identify_missing_paths()
-        s3_prefixes = set()
+        s3_buckets = set()
         for missing_path in missing_paths:
-            match = re.match(AWSResources.S3_PATH_REGEX, missing_path)
+            match = re.match(AWSResources.S3_BUCKET, missing_path)
             if match:
-                s3_prefixes.add(missing_path)
+                s3_buckets.add(match.group(1))
         if single_role:
-            roles.append(AWSUCRoleCandidate(role_name, policy_name, list(s3_prefixes)))
+            roles.append(AWSUCRoleCandidate(role_name, policy_name, list(s3_buckets)))
         else:
-            for idx, s3_prefix in enumerate(sorted(list(s3_prefixes))):
+            for idx, s3_prefix in enumerate(sorted(list(s3_buckets))):
                 roles.append(AWSUCRoleCandidate(f"{role_name}_{idx+1}", policy_name, [s3_prefix]))
         return roles
 
     def create_uc_roles(self, roles: list[AWSUCRoleCandidate]):
         roles_created = []
         for role in roles:
-            if self._aws_resources.create_uc_role(role.role_name):
+            expanded_paths = set()
+            for path in role.resource_paths:
+                expanded_paths.add(path)
+                expanded_paths.add(f"{path}/*")
+            role_arn = self._aws_resources.create_uc_role(role.role_name)
+            if role_arn:
                 self._aws_resources.put_role_policy(
                     role.role_name,
                     role.policy_name,
-                    set(role.resource_paths),
+                    expanded_paths,
                     self._aws_account_id,
                     self._kms_key,
                 )
                 roles_created.append(role)
+        time.sleep(15)
```

Comment:
can we use a `from databricks.sdk.retries import retried` instead of a fixed sleep duration